src/main/java/com/dungeoncrawler/Javiarenka/shop/BuyingWeaponInterface.java

```diff
@@ -0,0 +1,7 @@
+package com.dungeoncrawler.Javiarenka.shop;
+
+import com.dungeoncrawler.Javiarenka.equipment.Weapon;
+
+public interface BuyingWeaponInterface {
+    public Weapon getNewWeapon(); // todo ItemDropped item
+}
```

Comment:
Skoro Weapon i Armor dziedziczą po tej samej klasie Equipment, to możemy zrobić jeden interfejs do kupowania wszystkich przedmiotów przyjmujący obiekty Equipment, a w środku sobie rozkminimy, czy wpadł nam Armor czy Weapon. Ja podobnie zrobiłem metodę verifyAndAddStartingEquipment() w klasie CharacterCreatorService - przyjmuje Equipmenty, a potem rozkminia czy to armory czy weapony

Reply:
Zgadzam się. Zwłaszcza jak w przyszłości będziemy chcieli to rozwinąć np. jeszcze o pierścionki, paski czy inne elementy equipment.